### PR TITLE
docs: Document Overview Data Pipeline

### PR DESCRIPTION
**Pull request by kwankoravich**
## Why this PR
Why we need this PR?

## Changes
- Write some changes here

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/data/README.md [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+
+## Description
+
+The OpenThaiGPT Overview dataset pipeline is meticulously designed to fully preprocess raw data, extract essential information, and prepare the dataset in a trainable format. Here's an extended breakdown of its key components:
+
+## Workflow
+
+1. Obtain Raw Data in various formats such as CSV, PDF, etc., and convert it into plain text format.
+2. Transform plain text data from each raw data source into the JSONL format.
+3. Consolidate all JSONL datasets into a single file. This pipeline can be found at: `src/data/scripts/merge_jsonl/` to merge JSONL file
```
Comment (new5558) [inline src/data/README.md:17]: Only some data such as oscar, require merge jsonl. When we merge jsonl we do it seperately for each data.


For example
- oscar
    - oscar22.jsonl
    - oscar23.jsonl
    - oscar19.jsonl
- pantip
    - pantip2g.jsonl
    - pantip3g.jsonl
    

we need to run merge jsonl code two time targeting pantip and oscar seperately.


The result will be:
- oscar
    - oscar22.jsonl
    - oscar23.jsonl
    - oscar19.jsonl
    - oscarall.jsonl
- pantip
    - pantip2g.jsonl
    - pantip3g.jsonl
    - pantipall.jsonl
    - 
    
This is ugly, but it is our current pipeline works

File: src/data/README.md [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+4. Partition the JSONL dataset into training, validation, and test datasets. This pipeline can be found at: `src/data/scripts/split_data` to split the data into train/test/eval
+5. Anonymize Sensitive Data, which may include personal and residential information, in compliance with PDPA regulations.
```
Comment (new5558) [inline src/data/README.md:19]: No we anonymize last (after decontaminate)

and we also need to convert jsonl to huggingface dataset first before deduplicate starts https://github.com/OpenThaiGPT/openthaigpt-pretraining/tree/main/src/data/scripts/huggingface_create

File: src/data/README.md (hunk 1 above)
Comment (new5558) [reply src/data/README.md:17]: Also please add path to blind pdpa code https://github.com/OpenThaiGPT/openthaigpt-pretraining/tree/main/src/data/scripts/blind_pdpa

File: src/data/README.md [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
 Folder `notebook` contains the EDA or other experiment notebook.
 
 Folder `scripts` contains the main program which will call function in folder `openthaigpt_pretraining_data`
+
+## Description
+
+The OpenThaiGPT Overview dataset pipeline is meticulously designed to fully preprocess raw data, extract essential information, and prepare the dataset in a trainable format. Here's an extended breakdown of its key components:
+
+## Workflow
+
```
Comment (new5558) [inline src/data/README.md:14]: Please change from `src/data/scripts/…` to link to folder that contain the scripts on github

File: src/data/README.md [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
 
 Folder `notebook` contains the EDA or other experiment notebook.
 
 Folder `scripts` contains the main program which will call function in folder `openthaigpt_pretraining_data`
+
+## Description
+
+The OpenThaiGPT Overview dataset pipeline is meticulously designed to fully preprocess raw data, extract essential information, and prepare the dataset in a trainable format. Here's an extended breakdown of its key components:
+
+## Workflow
```
Comment (new5558) [inline src/data/README.md:13]: Can we have some diagram to explain the worlflow. We may export it from our figjam

**Review by new5558 — CHANGES_REQUESTED** [review]

File: src/data/README.md [hunk 5]
```diff
@@ -1,7 +1,23 @@
...
+
+1. Obtain Raw Data in various formats such as CSV, PDF, etc., and convert it into plain text format.
+2. Transform plain text data from each raw data source into the JSONL format.
```
Comment (new5558) [inline src/data/README.md:16]: We also have scripts for this krub

not all script is already in git for now, but the one you should include in this readme are: https://github.com/OpenThaiGPT/openthaigpt-pretraining/tree/main/src/data/scripts/pantip_2G
https://github.com/OpenThaiGPT/openthaigpt-pretraining/tree/main/src/data/scripts/pantip_3G
https://github.com/OpenThaiGPT/openthaigpt-pretraining/tree/main/src/data/scripts/internet

File: src/data/README.md [hunk 6]
```diff
@@ -1,7 +1,23 @@
...
 
 Folder `scripts` contains the main program which will call function in folder `openthaigpt_pretraining_data`
+
+## Description
+
+The OpenThaiGPT Overview dataset pipeline is meticulously designed to fully preprocess raw data, extract essential information, and prepare the dataset in a trainable format. Here's an extended breakdown of its key components:
+
+## Workflow
+
+1. Obtain Raw Data in various formats such as CSV, PDF, etc., and convert it into plain text format.
```
Comment (new5558) [inline src/data/README.md:15]: Not all data obtaining method are listed on git for now, but at least we should mentioned https://github.com/OpenThaiGPT/openthaigpt-pretraining/tree/main/src/data/scripts/crawl_thaigov in this readme

**Review by new5558 — APPROVED** [review]

**Review by boat1603 — APPROVED** [review]

**Pull request merged**